packages/expo-haptics/android/src/main/java/expo/modules/haptics/arguments/HapticsVibrationType.kt

```diff
@@ -0,0 +1,27 @@
+package expo.modules.haptics.arguments
+
+data class HapticsVibrationType(
+  val timings: LongArray,
+  val amplitudes: IntArray,
+  val oldSDKPattern: LongArray
+) {
+  override fun equals(other: Any?): Boolean {
+    if (this === other) return true
+    if (javaClass != other?.javaClass) return false
+
+    other as HapticsVibrationType
+
+    if (!timings.contentEquals(other.timings)) return false
+    if (!amplitudes.contentEquals(other.amplitudes)) return false
+    if (!oldSDKPattern.contentEquals(other.oldSDKPattern)) return false
+
+    return true
+  }
+
+  override fun hashCode(): Int {
+    var result = timings.contentHashCode()
+    result = 31 * result + amplitudes.contentHashCode()
+    result = 31 * result + oldSDKPattern.contentHashCode()
+    return result
+  }
```

Comment:
Why do we need this? 🤔 
Are we using this class for comparison/storing anywhere in the code? 🤔

Reply:
Because of the Array property in the data class, it's recommended to override the `equals` method.